### ISSUE TITLE
BlockFormat fails with "return macro"

### ISSUE THREAD

**Issue by Gama11** [adireddy/haxe-checkstyle#42]
``` haxe
class Macro
{
    function build()
    {
        return macro
        {
            while (true)
                trace("");
        }
    }
}
```

Results in:

```
./source/Macro.hx:7:4: warning: First line of multiline block should contain only {
./source/Macro.hx:7:4: warning: Last line of multiline block should contain only } and maybe , or ;
./source/Macro.hx:7:4: warning: First line of multiline block should contain only {
./source/Macro.hx:7:4: warning: Last line of multiline block should contain only } and maybe , or ;
```

I don't think this should fail. No warnings reported without the `macro`, so I guess that's where the issue lies.

Checkstyle version: 191bf0c245

**Comment by AlexHaxe** [comment]
BlockFormatCheck is still in the current dev and master branches, but it was replaced by Left- and RightCurlyChecks in the new token-tree branch. So you might want to that a try.

BlockFormatCheck (like many others) relies on AST information returned by haxeparse. I have not looked into it, but it might be hard to actually fix those checks, depending on how haxeparse handles macro statements.
